src/ncp/ncp_uart.cpp

```diff
@@ -279,6 +279,11 @@ void otPlatLog(otLogLevel aLogLevel, otLogRegion aLogRegion, const char *aFormat
     va_start(args, aFormat);
     if ((charsWritten = vsnprintf(logString, sizeof(logString), aFormat, args)) > 0)
     {
+        if (charsWritten > static_cast<int>(sizeof(logString) - 1))
```

Comment:
I actually had this code in my original but removed it. My understanding of this API is that it will return `sizeof(logString)` if it truncates, and it won't include a `NULL` terminator. Even in that case, `otNcpStreamWrite` doesn't need a `NULL` terminator because it just writes a byte array essentially. Notice in `NcpUart::HandleError` it also doesn't write the `NULL` terminator.